proof.go

```diff
@@ -162,7 +170,7 @@ func (proof Proof) verifyLeafHashes(nth *Hasher, verifyCompleteness bool, nID na
 				return false
 			}
 		}
-		// rightSubtrees only contains the subtrees after [0, r.Start)
+		// rightSubtrees only contains the subtrees after [0, r.end)
```

Comment:
oof, great find. Maybe we should use capitalisation here consistently? `[0, r.End)`